packages/openapi-generator/src/parser/unique-naming.ts

```diff
@@ -1,24 +1,75 @@
-import { UniqueNameGenerator, camelCase } from '@sap-cloud-sdk/util';
+import { UniqueNameGenerator, camelCase, unique } from '@sap-cloud-sdk/util';
+import { ParserOptions } from './options';
 
 /**
- * Ensure uniqueness of names.
- * Takes a list of names and renames those that are duplicate.
- * The renamed list has the same order as the original list.
- * @param names List of names to deduplicate.
- * @param options Object containing options to configure the transformation.
+ * Format names and ensure uniqueness of names.
+ * With `strictNaming` enabled, duplicate names lead to an error.
+ * With `strictNaming` disabled, duplicate names are renamed.
+ * The formatted (and renamed) list of names has the same order and length as the original list.
+ * @param names List of names to ensure uniqueness for.
+ * @param options Parser options.
+ * @param namingOptions Object containing options to configure the transformation.
  * @param options.format Function to format the name. Defaults to camel case.
  * @param options.reservedWords Reserved words that should be handled as duplicates.
  * @returns Unique names in the given order.
  */
 export function ensureUniqueNames(
   names: string[],
-  options: {
+  options: ParserOptions,
+  namingOptions: {
     format?: (name: string) => string;
     reservedWords?: string[];
   } = {}
 ): string[] {
-  const { format: format = camelCase, reservedWords = [] } = options;
+  const { format = camelCase, reservedWords = [] } = namingOptions;
```

Comment:
As default value, why not moving it to line 22, where `{}` is used as default `namingOptions`.

Reply:
Because I don't know the syntax for it if even possible.
If I move it up and do this: `{ format: camelCase, reservedWords: [] }` it will only be used if no `namingOptions` parameter was provided.
Moving it up with `{ format = camelCase, reservedWords = [] }` doesn't work.

I think it doesn't work, unless you know something maybe?

Reply:
I think you are right. If one parameter (`namingOptions`) is used, the default value is only applicable for this parameter and cannot be flexible enough for setting two properties (format/reservedWords) of the parameter.

Reply:
Schade.